Update Add-Ons Manager for Protocol Version 5

Noordfrees opened the pull request:
Counterpart to https://github.com/widelands/wl_addons_server/pull/95, see there for the list of new features

Fixes #4828 
Fixes #4873

Noordfrees committed:
Refactor `ui_fsmenu/addons.cc`

Noordfrees committed:
3 files were automatically formatted.

Noordfrees committed:
Merge branch 'master' of github.com:widelands/widelands into refactor_addons_cc

Noordfrees committed:
Use pgettext for some ambiguous strings

Noordfrees committed:
Some more

Noordfrees committed:
Merge branch 'master' of github.com:widelands/widelands into refactor_addons_cc

Noordfrees committed:
Show progress while refreshing remotes, and change Ctrl default behaviour in category icons

Fixes #4828
Re #4873

Noordfrees committed:
Optimize dependency error updating

Noordfrees committed:
Update comments

Noordfrees committed:
typo

Noordfrees committed:
Merge branch 'master' of github.com:widelands/widelands into addons-ui-enhancements

Noordfrees committed:
Merge branch 'master' of github.com:widelands/widelands into addons-ui-enhancements

Noordfrees committed:
Better status strings

Noordfrees committed:
Merge branch 'master' into addons-ui-enhancements

Noordfrees committed:
Merge branch 'master' into addons-ui-enhancements

Noordfrees committed:
Merge branch 'master' of github.com:widelands/widelands into addons-ui-enhancements

Noordfrees committed:
Update for protocol version 5, first round

Noordfrees committed:
Allow deleting comments

Noordfrees committed:
Update all features

Noordfrees committed:
Update doc

Noordfrees committed:
6 files were automatically formatted.

Noordfrees committed:
Suppress clang-tidy false positive

hessenfarmer commented:
can this be tested without the other branch? else I don't know how to help here

Noordfrees commented:
I can set up an alpha server to run in parallel to the official add-ons server. This might take five minutes or until tomorrow depending how much work setting up a separate alpha database (and its own docker container) is. Will report here when it's up for testing

Noordfrees committed:
Fix invalid memory access

Noordfrees committed:
Merge branch 'master' of github.com:widelands/widelands into addons-ui-enhancements

Noordfrees commented:
Alpha is up :) You can connect to it using the command-line switch `--addon_server_port=7377`
(don't forget to remove this from the config file when you want to use the real server again)

The alpha server runs in an encapsulated environment so you can write comments, add votes, submit content etc without affecting the official server and the server repo in any way. The alpha server will not perform any GitHub or Transifex integration at all.
I initialized the quality column with purely random values just for testing.

You can also test that you can connect to the alpha server from master without any problems, to check that backwards compatibility is working.

hessenfarmer commented:
ok I will try to test this tomorrow or on sunday

Noordfrees committed:
Merge branch 'master' into addons-ui-enhancements

matthiakl commented:
During the TX integration setup the client is unresponsive. A window should inform the user (admin), that the command is running.

Noordfrees committed:
Tweak Tx setup message

Noordfrees commented:
I added a note that this will take some time, and also a reminder to do the Transifex-UI maintenance that can annoyingly not be done via the commandline. (but maybe via an API call from the server? got to investigate this possibility sometime…)

Noordfrees committed:
'src/ui_fsmenu/addons/remote_interaction.cc' was automatically formatted.

Noordfrees committed:
Merge branch 'master' of github.com:widelands/widelands into addons-ui-enhancements

Noordfrees committed:
Fix comment row layouting glitch

Noordfrees committed:
'src/ui_fsmenu/addons/remote_interaction.cc' was automatically formatted.

matthiakl reviewed (commented):
Does the `all` argument of CMD_LIST currently has any effect or use case? From what I see, incompatible Add-Ons are filtered either by the server or by the client.
For the comment index you use a string now, but I think it should stay a number.

Noordfrees committed:
Merge branch 'master' of github.com:widelands/widelands into addons-ui-enhancements

Noordfrees committed:
Address review and fix another glitch

Noordfrees commented:
Changed the comment indices to `int` and changed the filtering so that Ctrl-clicking "Refresh" always fetches (and displays) all remotes while normal click fetches only compatible ones, how about that?

Noordfrees committed:
2 files were automatically formatted.

matthiakl reviewed (approved):
> Ctrl-clicking "Refresh" always fetches (and displays) all remotes while normal click fetches only compatible ones, how about that?

Fine for now, however in the future a visible selector (like for verified add-ons) should be implemented, IMHO.
The version comparison seems to have an issue. With this branch (version `1.1~git25482[656516b@review]`) the add-on legacy_terrains only shows up with Ctrl+click, however 1.1 should be > 1.0.99.

Rest LGTM now, maybe @hessenfarmer  had some time for testing as well?

Noordfrees commented:
> The version comparison seems to have an issue. With this branch (version `1.1~git25482[656516b@review]`) the add-on legacy_terrains only shows up with Ctrl+click, however 1.1 should be > 1.0.99.

Widelands does not display the mismatch as an error so this must be a bug in the server. Will investigate
Edit: One-liner fix, added it to https://github.com/widelands/wl_addons_server/pull/106

hessenfarmer commented:
No I did not find any time to test this however this will be tested pretty much until the next release I believe, so no objections from my side against merging
